### PR TITLE
Add Plonkup

### PR DESCRIPTION
**Pull request by vlasin**

### CODE REVIEW

File: src/ZkFold/Cardano/OffChain/Transcript.hs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    toTranscript (F n) = integerToByteString LittleEndian 32 n
+
+instance ToTranscript BuiltinByteString G1 where
+    toTranscript = bls12_381_G1_compress
+
+instance FromTranscript BuiltinByteString F where
+    fromTranscript = F . byteStringToInteger LittleEndian . blake2b_224
+
+instance ToTranscript BuiltinByteString Fr where
+    toTranscript = integerToByteString LittleEndian 32 . convertZp
```
Comment (zlonast) [inline src/ZkFold/Cardano/OffChain/Transcript.hs:31]: I can't understand why we have F and Fr. Isn't it the same thing?

File: src/ZkFold/Cardano/OffChain/ECC.hs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module ZkFold.Cardano.OffChain.ECC where
```
Comment (zlonast) [inline src/ZkFold/Cardano/OffChain/ECC.hs:1]: Perhaps we should write a comment on how the abbreviation stands. ECC is Elliptic-curve cryptography

**Review by zlonast — APPROVED** [review]

File: src/ZkFold/Cardano/OffChain/Transcript.hs (hunk 1 above)
Comment (vlasin) [reply src/ZkFold/Cardano/OffChain/Transcript.hs:31]: No, they are not. `F` is defined in `zkfold-cardano` and is based on `Integer` while `Fr` is based on `Natural` and is defined in `zkfold-base`.

Comment (vlasin) [reply src/ZkFold/Cardano/OffChain/Transcript.hs:31]: So, we can't use `Fr` in our on-chain code directly. Plus, in the future `Fr` might be based on `ByteString`.

Comment (zlonast) [reply src/ZkFold/Cardano/OffChain/Transcript.hs:31]: Thank you, now I see how it happened here

**Pull request merged**